src/node_i18n.cc

```diff
@@ -339,6 +343,70 @@ static void ICUErrorName(const FunctionCallbackInfo<Value>& args) {
                           v8::NewStringType::kNormal).ToLocalChecked());
 }
 
+#define TYPE_ICU "icu"
+#define TYPE_UNICODE "unicode"
+#define TYPE_CLDR "cldr"
+#define TYPE_TZ "tz"
+
+/**
+ * This is the workhorse function that deals with the actual version info.
+ * Get an ICU version.
+ * @param type the type of version to get. One of VERSION_TYPES
+ * @param buf optional buffer for result
+ * @param status ICU error status. If failure, assume result is undefined.
+ * @return version number, or NULL. May or may not be buf.
+ */
+static const char* GetVersion(const char* type,
+                              char buf[U_MAX_VERSION_STRING_LENGTH],
+                              UErrorCode* status) {
+  if (!strcmp(type, TYPE_ICU)) {
+    return U_ICU_VERSION;
+  } else if (!strcmp(type, TYPE_UNICODE)) {
+    return U_UNICODE_VERSION;
+  } else if (!strcmp(type, TYPE_TZ)) {
+    return TimeZone::getTZDataVersion(*status);
+  } else if (!strcmp(type, TYPE_CLDR)) {
+    UVersionInfo versionArray;
+    ulocdata_getCLDRVersion(versionArray, status);
+    if (U_SUCCESS(*status)) {
+      u_versionToString(versionArray, buf);
+      return buf;
+    }
+  }
+  // Fall through - unknown type or error case
+  return nullptr;
+}
+
+static void GetVersion(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  if ( args.Length() == 0 ) {
+    // With no args - return a comma-separated list of allowed values
+      args.GetReturnValue().Set(
+          String::NewFromUtf8(env->isolate(),
+            TYPE_ICU ","
+            TYPE_UNICODE ","
+            TYPE_CLDR ","
+            TYPE_TZ));
+  } else {
+    CHECK_GE(args.Length(), 1);
+    CHECK(args[0]->IsString());
+    Utf8Value val(env->isolate(), args[0]);
+    UErrorCode status = U_ZERO_ERROR;
+    char buf[U_MAX_VERSION_STRING_LENGTH] = "";  // Possible output buffer.
+    const char* versionString = GetVersion(*val, buf, &status);
+
+    if (U_SUCCESS(status) && versionString) {
+      // Success.
+      args.GetReturnValue().Set(
+          String::NewFromUtf8(env->isolate(),
+          versionString));
+    } else {
+      // Fail => undefined
```

Comment:
This shouldn't be necessary to set explicitly. The function will return undefined automatically if no return value is set.